storage/importexport/src/main/java/de/danoeh/antennapod/storage/importexport/AutomaticDatabaseExportWorker.java

```diff
@@ -0,0 +1,127 @@
+import de.danoeh.antennapod.ui.notifications.NotificationUtils;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Locale;
+import java.util.concurrent.TimeUnit;
+import org.greenrobot.eventbus.EventBus;
+
+public class AutomaticDatabaseExportWorker extends Worker {
+    private static final String WORK_ID_AUTOMATIC_DATABASE_EXPORT = "de.danoeh.antennapod.AutomaticDbExport";
+
+    public static void enqueueIfNeeded(Context context, boolean replace) {
+        if (UserPreferences.getAutomaticExportFolder() == null) {
+            WorkManager.getInstance(context).cancelUniqueWork(WORK_ID_AUTOMATIC_DATABASE_EXPORT);
+        } else {
+            PeriodicWorkRequest workRequest = new PeriodicWorkRequest.Builder(
+                        AutomaticDatabaseExportWorker.class, 1, TimeUnit.DAYS)
+                    .build();
+            WorkManager.getInstance(context).enqueueUniquePeriodicWork(WORK_ID_AUTOMATIC_DATABASE_EXPORT,
+                    replace ? ExistingPeriodicWorkPolicy.REPLACE : ExistingPeriodicWorkPolicy.KEEP, workRequest);
+        }
+    }
+
+    public AutomaticDatabaseExportWorker(@NonNull Context context, @NonNull WorkerParameters params) {
+        super(context, params);
+    }
+
+    @Override
+    @NonNull
+    public Result doWork() {
+        String folderUri = UserPreferences.getAutomaticExportFolder();
+        if (folderUri == null) {
+            return Result.success();
+        }
+        try {
+            export(folderUri);
+            return Result.success();
+        } catch (IOException e) {
+            showErrorNotification(e);
+            return Result.failure();
+        }
+    }
+
+    private void export(String folderUri) throws IOException {
+        DocumentFile documentFolder = DocumentFile.fromTreeUri(getApplicationContext(), Uri.parse(folderUri));
+        if (documentFolder == null || !documentFolder.exists() || !documentFolder.canWrite()) {
+            throw new IOException("Unable to open export folder");
+        }
+        String filename = String.format("AntennaPodBackup-%s.db",
+                new SimpleDateFormat("yyyy-MM-dd", Locale.US).format(new Date()));
+        DocumentFile exportFile = documentFolder.createFile("application/x-sqlite3", filename);
+        if (exportFile == null || !exportFile.canWrite()) {
+            throw new IOException("Unable to create export file");
+        }
+        DatabaseExporter.exportToDocument(exportFile.getUri(), getApplicationContext());
+        List<DocumentFile> files = new ArrayList<>(Arrays.asList(documentFolder.listFiles()));
+        Iterator<DocumentFile> itr = files.iterator();
+        while (itr.hasNext()) {
+            DocumentFile file = itr.next();
+            if (!file.getName().startsWith("AntennaPod")) {
```

Comment:
This feels a little unsafe, as the user may plausibly have files other than database backups in their chosen backup folder, such as the various HTML exports (I can see they're lowercase by default, but potentially could have been renamed at the manual save screen) or files relating to other backups (I have occasionally tweaked and reimported backup files to fix dodgy feeds or inaccurate play time/dates, and have some SQL files named after the source databases).

I am assuming `documentFolder.listFiles()` returns all files in the directory (as opposed to just the ones created by the current app) -- if so, short of doing full pattern matches against files present in the directory, this would be safer:

```java
if (!file.getName().startsWith("AntennaPodBackup-") || !file.getName().endsWith(".db")) {
```

For context, I enable automatic backups on any app that provides it, sending them to `/Backups` and syncing that off-device using Syncthing.  I could get around the app wanting to manage the chosen directory by giving it a subdirectory, but it's not exactly clear from the feature description ("Automatically create daily backups of the AntennaPod database") that it will also be deleting files outside of the latest it wants to keep.